library/Ivoz/Provider/Domain/Model/MatchList/MatchListDto.php

```diff
@@ -31,6 +45,28 @@ public static function getPropertyMap(string $context = '', string $role = null)
         return $response;
     }
 
+    /**
+     * @return array<array-key, mixed>
+     */
+    public function normalize(string $context, string $role = ''): array
+    {
+        $response = parent::normalize(
+            $context,
+            $role
+        );
+
+        if ($context === self::CONTEXT_COLLECTION) {
+            if ($role === 'ROLE_COMPANY_ADMIN') {
+                $editable = $this->getCompanyId() != null;
+                $this->setEditable($editable);
+                $response['editable'] = $editable;
+            }
+        }
```

Comment:
Psalm complains for attribute `$editable` not being used, and other classes have getters and setters for custom fields. I liked the idea but I was not sure what would be the best way to implement it.